backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/writer/impl/IndexWriterDelegatorImpl.java

```diff
@@ -173,15 +182,19 @@ void closeAfterFailure(Throwable throwable, Object failingOperation) {
 	}
 
 	private void doCommit() {
+		commitLock.lock();
 		try {
-			synchronized (commitLock) {
-				delegate.commit();
-				updateCommitExpiration();
-			}
+			// NOTE: underlying Lucene code is using same pattern to sync on object block:
```

Comment:
Nice to add a comment, but I think it needs a bit more context :)

```suggestion
			// NOTE: underlying Lucene code is using this pattern to sync on object block,
			// which could be a problem with Loom:
```